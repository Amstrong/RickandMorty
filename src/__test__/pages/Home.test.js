import React from "react";
import { mount } from "enzyme";
import { Link } from "react-router-dom";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router";
import Home from "../../frontend/pages/home/homePage";
import ProviderMock from "../../__mocks__/ProviderMock";

describe("<Home/>", () => {
  test("Home: Checking title", () => {
    const home = mount(
      <ProviderMock>
        <Home />
      </ProviderMock>
    );
    expect(home.find(".text--title").text()).toEqual("Rick&Morty");
  });

  test("Home: Verifying the ReactRouter Link", () => {
    const component = renderer.create(
      <StaticRouter location="someLocation" context={{}}>
        <Link to="/characters" />
      </StaticRouter>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
