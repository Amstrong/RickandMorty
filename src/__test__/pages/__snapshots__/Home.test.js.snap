// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Home: Verifying the ReactRouter Link 1`] = `
<a
  href="/characters"
  onClick={[Function]}
/>
`;
